sdcm/cluster.py

```diff
@@ -4071,19 +4071,30 @@ def node_setup(self, node: BaseNode, verbose: bool = False, timeout: int = 3600)
 
         if not Setup.REUSE_CLUSTER:
             node.disable_daily_triggered_services()
+
             # prepare and start saslauthd service
-            setup_script = dedent(f"""
-                sudo yum install -y cyrus-sasl
-                sudo systemctl enable saslauthd
-                echo 'MECH=ldap' | sudo tee -a /etc/sysconfig/saslauthd
-                sudo touch /etc/saslauthd.conf
-                echo 'saslauthd_socket_path: /run/saslauthd/mux' | sudo tee -a /etc/scylla/scylla.yaml
-            """)
-            node.remoter.run('bash -cxe "%s"' % setup_script)
-            conf = node.get_saslauthd_config()
-            for key in conf.keys():
-                node.remoter.run(f'echo "{key}: {conf[key]}" | sudo tee -a /etc/saslauthd.conf')
-            node.remoter.sudo('systemctl restart saslauthd')
+            if self.params.get('prepare_saslauthd'):
+                if node.is_rhel_like():
+                    setup_script = dedent(f"""
+                        sudo yum install -y cyrus-sasl
+                        sudo systemctl enable saslauthd
+                        echo 'MECH=ldap' | sudo tee -a /etc/sysconfig/saslauthd
+                        sudo touch /etc/saslauthd.conf
+                        echo 'saslauthd_socket_path: /run/saslauthd/mux' | sudo tee -a /etc/scylla/scylla.yaml
```

Comment:
These 2 lines may not necessarily be duplicated.

Reply:
> These 2 lines may not necessarily be duplicated.

The above two lines write two different configure files.

```
                        sudo touch /etc/saslauthd.conf
                        echo 'saslauthd_socket_path: /run/saslauthd/mux' | sudo tee -a /etc/scylla/scylla.yaml
```

Reply:
@yarongilor , I just moved the last statement out, and edit the scylla.yaml by remote_scylla_yaml()